### ISSUE TITLE
Rokus With AirPlay Are Not Discovered in AirPlay Player Providers

### ISSUE THREAD

**Issue by taydy00** [music-assistant/hass-music-assistant#2242]
### What version of Music Assistant has the issue?

2.0.0b145 (but has always had this issue)

### What version of the Home Assistant Integration have you got installed?

2024.5.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Hey all, I have multiple Roku 4K devices (with AirPlay and HomeKit built in). They are both part of my Apple Home (that I no longer use, but regardless they are still part of it). They show up as AirPlay devices for pretty much any Apple device on my (flat) network, and Home Assistant itself has them as media players. However, Music Assistant doesn't see them as AirPlay devices (but does successfully see my HomePod Mini, so AirPlay devices are working correctly). I read about some devices needing to have encryption enabled, however the Rokus don't even show in Music Assistant for me to enable/disable encryption. Is it possible to add player providers manually via IP?

I did also attempt to add them via the Home Assistant Media Player integration, however Roku disabled third-party "Play on Roku" support so that workaround won't work either. You will probably see this in the logs as well.

### How to reproduce

Try to add AirPlay Player Providers, notice that Rokus are not auto-discovered.

### Music Providers

All - this is unrelated to the issue (I believe). However, I have:

Spotify
Jellyfin (currently down, please ignore in logs)
Local Media

### Player Providers

AirPlay (though again, my HomePod Mini is auto-discovered and works as it should)

### Full log output

[log.txt](https://github.com/music-assistant/hass-music-assistant/files/15277083/log.txt)

### Additional information

My network is a flat network, and there is only one interface for Home Assistant. The only notable thing about the setup is that I use Tailscale (which runs on Home Assistant). However, all of my Linux boxes (including Home Assistant) are able to see all other devices on the LAN, so I don't believe that to be an issue.

### What version of Home Assistant Core are your running

2024.5.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Are these Airplay 1 or Airplay 2 devices? Airplay 2  only devices are not supported

**Comment by taydy00** [comment]
Ah, you would be correct, they are AirPlay 2. I misunderstood when it said that "AirPlay 2 devices should receive the AirPlay 1 stream without issues". So this means that we can't send an AirPlay 1 stream via MASS to an AirPlay 2 device?

**Comment by OzGav** [comment]
MA will do that but if the device won't/can't understand it then problem...

**Comment by OzGav** [comment]
If you look forther down on that page you will see Samsung does something weird and their devices don't work and not just a problem for MA

**Comment by taydy00** [comment]
I understand - so there's no way to get MASS to see the AirPlay 2 device and try to send a stream to it? Is there anyway I can manually create a playback provider/device just to see what happens?

**Comment by taydy00** [comment]
Right, for sure, but at least the Samsung devices show up within Music Assistant. I can't even see the Roku devices within Music Assistant, hence why I'm wondering if there's something else at play here?

**Comment by OzGav** [comment]
You can check the device settings. If your mini is on the same subnet then there should be no reason why it doesn't work. Have you tried encryption on/off and compression on/off?

**Comment by taydy00** [comment]
Sorry if I wasn't clear - The HomePod Mini works just fine and is discovered by Music Assistant. However, my Rokus (with AirPlay 2) are not discovered by Music Assistant. I can't change the encryption/compression settings for the Rokus until Music Assistant can see them, but it does not discover the Rokus. Ultimately, I'm wondering if I can manually add the Roku devices to Music Assistant

**Comment by OzGav** [comment]
Oh of course. I wasn't thinking. No you can't manually add devices. I'll move this to a feature request for now and speak to Marcel in due course. He has mentioned trying to get Airplay 2/ATV devices working. Priority unknown!
